File: Baekjoon/Class4/16.2096 내려가기.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

const TABLE_SIZE = 3;

const n = Number(input[0]);

let prevMaxTable = new Array(3).fill(0);
let prevMinTable = new Array(3).fill(0);
let maxTable = input[1].split(" ").map(Number); // 첫 열
let minTable = [...maxTable];

for(let i = 2; i <= n; i++) {
    prevMaxTable = [...maxTable]; // 슬라이딩 윈도우
    prevMinTable = [...minTable];
    maxTable = input[i].split(" ").map(Number); // 다음 열
    minTable = [...maxTable];

    maxTable[0] += Math.max(prevMaxTable[0], prevMaxTable[1]); // 최댓값 누적
    maxTable[1] += Math.max(...prevMaxTable);
    maxTable[2] += Math.max(prevMaxTable[1], prevMaxTable[2]);
    minTable[0] += Math.min(prevMinTable[0], prevMinTable[1]); // 최솟값 누적
    minTable[1] += Math.min(...prevMinTable);
    minTable[2] += Math.min(prevMinTable[1], prevMinTable[2]);
}

console.log(`${Math.max(...maxTable)} ${Math.min(...minTable)}`);

File: Baekjoon/Class4/16.2096 내려가기.py
import sys
input = sys.stdin.readline

TABLE_SIZE = 3

n = int(input())

prev_max_table = None
prev_min_table = None
max_table = list(map(int, input().split())) #첫 열
min_table = max_table[:]

for i in range(1, n):
    prev_max_table = max_table[:] #슬라이딩 윈도우
    prev_min_table = min_table[:]
    max_table = list(map(int, input().split())) #다음 열
    min_table = max_table[:]

    max_table[0] += max(prev_max_table[0], prev_max_table[1]) #최댓값 누적
    max_table[1] += max(prev_max_table)
    max_table[2] += max(prev_max_table[1], prev_max_table[2])
    min_table[0] += min(prev_min_table[0], prev_min_table[1]) #최솟값 누적
    min_table[1] += min(prev_min_table)
    min_table[2] += min(prev_min_table[1], prev_min_table[2])

print(f"{max(max_table)} {min(min_table)}")